File: src/main/java/leets/leenk/domain/media/domain/service/MediaDeleteService.java
package leets.leenk.domain.media.domain.service;

import leets.leenk.domain.feed.domain.entity.Feed;
import leets.leenk.domain.media.domain.entity.Media;
import leets.leenk.domain.media.domain.repository.MediaRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
@RequiredArgsConstructor
public class MediaDeleteService {

    private final MediaRepository mediaRepository;

    public void delete(Media media) {
        mediaRepository.delete(media);
    }

    public void deleteAll(List<Media> mediaList) {
        mediaRepository.deleteAll(mediaList);
    }

    public void deleteAllByFeed(Feed feed) {
        mediaRepository.deleteAllByFeed(feed);
        mediaRepository.flush();
    }
}

File: src/main/kotlin/leets/leenk/domain/media/domain/service/MediaDeleteService.kt
package leets.leenk.domain.media.domain.service

import leets.leenk.domain.feed.domain.entity.Feed
import leets.leenk.domain.media.domain.entity.Media
import leets.leenk.domain.media.domain.repository.MediaRepository
import org.springframework.stereotype.Service

@Service
class MediaDeleteService(
    private val mediaRepository: MediaRepository,
) {
    fun delete(media: Media) {
        mediaRepository.delete(media)
    }

    fun deleteAll(mediaList: List<Media>) {
        mediaRepository.deleteAll(mediaList)
    }

    fun deleteAllByFeed(feed: Feed) {
        mediaRepository.deleteAllByFeed(feed)
        mediaRepository.flush()
    }
}
